testing/internal/inventorysorter/pkgmatcher.go

```diff
@@ -0,0 +1,37 @@
+// Package inventorysorter provides a Sort function for inventories to allow
+// clear comparisons with cmp.Diff()
+package inventorysorter
+
+import (
+	"cmp"
+	"slices"
+	"strings"
+
+	"github.com/google/osv-scalibr/extractor"
+)
+
+// Sort sorts the inventories in-place to allow cmp matching
+// This does not sort all available fields (e.g. Metadata)
+func Sort(inv []*extractor.Inventory) {
```

Comment:
If this is for testing why not just sort based on GetPurl()? That should include most of the metadata being compared.

Reply:
Hmm might work, it does only sort for Name and Version, but that should be fine in almost all cases, since we dedup on those anyway in osv-scanner extractors. 

Though I think this will fail if we apply this to the current requirements.txt implementation, since the same package can appear in multiple locations.

Reply:
While looking into this, I realized the current ToPURL() functions are actually incorrect, it is missing the conversion from `_` to `-` that the spec says is required: https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#pypi

I united all 5 python extractors to use the same internal ToPURL function, since the implementation is identical, and in that function implemented both transformations that the spec requires.

Reply:
I think we can update this when we switch the tests for requirements.txt - it doesn't use maps at the moment so it doesn't really need sorting right now. So for this PR I think it's fine to switch to using PURLs to sort.

Reply:
Updated to use PURL to sort, let me know if you think I should remove the inventorysorter package and just place it in extracttest, or keep it as is.

Reply:
Yeah, I'd say the function is short enough that we can just inline it in extracttest.

Reply:
Done